tests/UnifiedSpecTests/Constraint/MatchesTest.php

```diff
@@ -23,6 +23,14 @@ public function testMatchesDocument(): void
         $this->assertResult(true, $c, ['y' => ['b' => 2, 'a' => 1], 'x' => 1], 'Root and embedded key order is not significant');
     }
 
+    public function testFlexibleNumericComparison(): void
```

Comment:
Added these for the sake of completeness (and because I wanted to make sure that this works in the unified test runner).